shotover-proxy/tests/opensearch_int_tests/mod.rs

```diff
@@ -1,30 +1,27 @@
+use crate::shotover_process;
 use opensearch::{
     auth::Credentials,
     cert::CertificateValidation,
     http::Url,
     http::{
-        response::Response,
         transport::{SingleNodeConnectionPool, TransportBuilder},
         StatusCode,
     },
-    indices::IndicesExistsParts,
+    indices::{IndicesCreateParts, IndicesDeleteParts, IndicesExistsParts},
     params::Refresh,
-    BulkOperation, BulkParts, OpenSearch, SearchParts,
+    BulkOperation, BulkParts, DeleteParts, IndexParts, OpenSearch, SearchParts,
 };
 use serde_json::{json, Value};
 use test_helpers::docker_compose::docker_compose;
 
-pub async fn index_documents(client: &OpenSearch) -> Response {
-    let index = "posts";
-    let exists_response = client
+pub async fn test_bulk(client: &OpenSearch) {
```

Comment:
If we sort our search results then we can assert that we get the exact list of items that we expect:
```rust
pub async fn test_bulk(client: &OpenSearch) {
    assert_ok_and_get_json(
        client
            .indices()
            .create(IndicesCreateParts::Index("posts"))
            .send()
            .await,
    )
    .await;

    let mut body: Vec<BulkOperation<_>> = vec![];
    for i in 0..10 {
        let op = BulkOperation::index(json!({"title": "OpenSearch", "i": i}))
            .id(i.to_string())
            .into();
        body.push(op);
    }

    assert_ok_and_get_json(
        client
            .bulk(BulkParts::Index("posts"))
            .body(body)
            .refresh(Refresh::WaitFor)
            .send()
            .await,
    )
    .await;

    let results = assert_ok_and_get_json(
        client
            .search(SearchParts::None)
            .body(json!({
                "query": {
                    "match_all": {}
                },
                "sort": [
                    {
                        "i": {
                            "order": "asc"
                        }
                    }
                ]
            }))
            .allow_no_indices(true)
            .send()
            .await,
    )
    .await;

    assert!(results["took"].is_i64());
    let hits = results["hits"]["hits"].as_array().unwrap();
    assert_eq!(
        hits.iter().map(|x| &x["_source"]).collect::<Vec<_>>(),
        vec!(
            &json!({ "title": "OpenSearch", "i": 0 }),
            &json!({ "title": "OpenSearch", "i": 1 }),
            &json!({ "title": "OpenSearch", "i": 2 }),
            &json!({ "title": "OpenSearch", "i": 3 }),
            &json!({ "title": "OpenSearch", "i": 4 }),
            &json!({ "title": "OpenSearch", "i": 5 }),
            &json!({ "title": "OpenSearch", "i": 6 }),
            &json!({ "title": "OpenSearch", "i": 7 }),
            &json!({ "title": "OpenSearch", "i": 8 }),
            &json!({ "title": "OpenSearch", "i": 9 }),
        )
    );
}
```